proposed/http-middleware/middleware.md

```diff
@@ -1,114 +1,123 @@
 HTTP Server Middleware
```

Comment:
Should the name of the proposal be updated, since we cover both middleware and request handlers? Perhaps "HTTP Server Components" (to borrow verbiage from the opening paragraph)?